common/exception/src/exception.rs

```diff
@@ -45,6 +51,139 @@ impl ToString for ErrorCodeBacktrace {
         }
     }
 }
+/// Put Sled's errors wrapper here to avoid circular references
+/// Error returned by a Sled transaction
+#[derive(Debug)]
+pub enum SledTransactionError<T> {
+    Abort(T),
+    /// A storage related error
+    Storage(io::Error),
+}
```

Comment:
`#[derive(this_error::Error)]` would save a lot lines of code for you 🤔

Reply:
Must flush a :rocket:  to you @drmingdrmer